addons/l10n_sg_ubl_pint/models/account_edi_xml_pint_sg.py

```diff
@@ -0,0 +1,108 @@
+                'tax_amount': taxes_vals['tax_amount'],
+                'tax_subtotal_vals': [],
+            }
+            for _grouping_key, vals in taxes_vals['tax_details'].items():
+                tax_totals_vals['tax_subtotal_vals'].append({
+                    'currency': company_currency,
+                    'currency_dp': company_currency.decimal_places,
+                    'taxable_amount': vals['base_amount'],
+                    'tax_amount': vals['tax_amount'],
+                    'percent': vals['_tax_category_vals_']['percent'],
+                    'tax_category_vals': vals['_tax_category_vals_'],
+                })
+            vals_list.append(tax_totals_vals)
+        return vals_list
+
+    def _get_tax_category_list(self, invoice, taxes):
+        # EXTENDS account_edi_ubl_cii
+        vals_list = super()._get_tax_category_list(invoice, taxes)
+        for vals in vals_list:
+            vals['tax_scheme_vals'] = {'id': 'GST'}
+        return vals_list
+
+    def _get_customization_ids(self):
+        # EXTENDS account_edi_ubl_cii
+        vals = super()._get_customization_ids()
+        vals['pint_sg'] = 'urn:peppol:pint:billing-1@sg-1'
+        return vals
+
+    def _export_invoice_vals(self, invoice):
+        # EXTENDS account_edi_ubl_cii
+        vals = super()._export_invoice_vals(invoice)
+
+        vals.update({
+            'InvoiceType_template': 'l10n_sg_ubl_pint.ubl_21_InvoiceType_sg',
+            'CreditNoteType_template': 'l10n_sg_ubl_pint.ubl_21_CreditNoteType_sg',
+            'DebitNoteType_template': 'l10n_sg_ubl_pint.ubl_21_CreditNoteType_sg',
+        })
+
+        vals['vals'].update({
+            # see https://docs.peppol.eu/poac/sg/2024-Q2/pint-sg/bis/#_bis_identifiers
+            'customization_id': self._get_customization_ids()['pint_sg'],
+            'profile_id': 'urn:peppol:bis:billing',
+            'uuid': invoice._l10n_sg_get_uuid(),
+        })
+
+        if invoice.currency_id != invoice.company_id.currency_id:
+            # see https://docs.peppol.eu/poac/sg/2024-Q2/pint-sg/bis/#_invoice_totals_in_gst_accounting_currency
+            vals['vals']['tax_currency_code'] = invoice.company_id.currency_id.name  # accounting currency
+        return vals
+
+    def _export_invoice_constraints(self, invoice, vals):
+        # EXTENDS account_edi_ubl_cii
+        constraints = super()._export_invoice_constraints(invoice, vals)
+
+        # Tax category must be filled on the line, with a value from SG categories.
+        for tax_total_val in vals['vals']['tax_total_vals']:
+            for tax_subtotal_val in tax_total_val['tax_subtotal_vals']:
+                tax_category_vals = tax_subtotal_val['tax_category_vals']
+                if tax_category_vals['tax_category_code'] not in SG_TAX_CATEGORIES:
+                    constraints['sg_vat_category_required'] = _("You must set a Singaporean tax category on each taxes of the invoice.")
```

Comment:
we need a pot file + the po file for the target language needed

Reply:
No idea what language is needed. l10n_sg is not translated too.
I think english is one of the official languages, so it should be enough for now.

Reply:
we still need a pot file :grin:, even if we don't add any po file